internal/cmd/cmd_zadd.go

```diff
@@ -0,0 +1,207 @@
+package cmd
+
+import (
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/dicedb/dice/internal/errors"
+	"github.com/dicedb/dice/internal/eval/sortedset"
+	"github.com/dicedb/dice/internal/object"
+	"github.com/dicedb/dice/internal/shardmanager"
+	dsstore "github.com/dicedb/dice/internal/store"
+)
+
+var cZADD = &CommandMeta{
+	Name:      "ZADD",
+	Syntax:    "ZADD key [NX | XX] [GT | LT] [CH] [INCR] score member [score member...]",
+	HelpShort: "Adds all the specified members with the specified scores to the sorted set stored at key",
+	HelpLong: `
+Adds all the specified members with the specified scores to the sorted set stored at key
+Options: NX, XX, CH, INCR, GT, LT, CH
+- NX: Only add new elements and do not update existing elements
+- XX: Only update existing elements and do not add new elements
+- CH: Modify the return value from the number of new elements added, to the total number of elements changed
+- INCR: When this option is specified, the elements are treated as increments to the score of the existing elements
+- GT: Only add new elements if the score is greater than the existing score
+- LT: Only add new elements if the score is less than the existing score
+Returns the number of elements added to the sorted set, not including elements already existing for which the score was updated.
+	`,
+	Examples: `
+localhost:7379> ZADD mySortedSet 1 foo 2 bar
+OK 2
+`,
+	Eval:    evalZADD,
+	Execute: executeZADD,
+}
+
+func init() {
+	CommandRegistry.AddCommand(cZADD)
+}
+
+func evalZADD(c *Cmd, s *dsstore.Store) (*CmdRes, error) {
+	if len(c.C.Args) < 3 {
+		return cmdResNil, errors.ErrWrongArgumentCount("ZADD")
+	}
+	key := c.C.Args[0]
+	sortedSet, err := getOrCreateSortedSet(s, key)
+	if err != nil {
+		return cmdResNil, err
+	}
+	// flags parsing
+	flags, nextIndex := parseFlags(c.C.Args[1:])
+	if nextIndex >= len(c.C.Args) || (len(c.C.Args)-nextIndex)%2 != 0 {
```

Comment:
Nit: Shall we use `nextIndex` to avoid the processing of flags?
> If nextIndex is 1 (which would be the position right after the key in the command arguments), that means no flag arguments were processed and we can skip the flag parsing entirely.

Reply:
I didn't understand this one. Addressed others.